<img src="../imagen/encabezado.png" style="width:auto;height:auto;">

<h1 align="center">Regresor para aplicaciones moviles</h1>

<h2>Indice</h2>
<ol>
    <li>Librerias</li>
    <li>Conjunto de Datos</li>
    <li>Preprocesamiento
        <ol>
            <li>Renombrar Caracteristicas</li>
            <li>Eliminar Caracteristicas</li>
            <li>Conversión de Datos</li> <!-- estos dos ultimos pueden ser que cambien de orden-->
            <li>Eliminar Registros</li>
        </ol>
    </li>
    <li>Procesamiento
        <ol>
            <li></li>
        </ol>
    </li>
</ol>

<h1>Librerias</h1>

In [134]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

<h1 id="uno"> CONJUNTO DE DATOS </h1>

Dirección del conjunto de datos original: https://www.kaggle.com/lava18/google-play-store-apps

El conjunto de datos original tiene 13 caracteristicas con 10841 registros.

In [135]:
d = pd.read_csv('../dataset/app.csv.zip')

In [136]:
d

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
5                             Paper flowers instructions       ART_AND_DESIGN   
6                Smoke Effect Photo Maker - Smoke Editor       ART_AND_DESIGN   
7                                       Infinite Painter       ART_AND_DESIGN   
8                                   Garden Coloring Book       ART_AND_DESIGN   
9                          Kids Paint Free - Drawing Fun       ART_AND_DESIGN   
10                               Text on Photo - Fonteee       ART_AND_DESIGN   
11               Name Art Photo Editor - Focus n Filters       ART_AND_DESIGN   
12                        Tattoo Name On My Photo Editor       ART_AND_DESIGN   
13                                 Mandala Coloring Book       ART_AND_DESIGN   
14       3D Color Pixel by Number - Sandbox Art Coloring       ART_AND_DESIGN   
15                       Learn To Draw Kawaii Characters       ART_AND_DESIGN   
16          Photo Designer - Write your name with shapes       ART_AND_DESIGN   
17                              350 Diy Room Decor Ideas       ART_AND_DESIGN   
18                         FlipaClip - Cartoon animation       ART_AND_DESIGN   
19                                          ibis Paint X       ART_AND_DESIGN   
20                           Logo Maker - Small Business       ART_AND_DESIGN   
21             Boys Photo Editor - Six Pack & Men's Suit       ART_AND_DESIGN   
22               Superheroes Wallpapers | 4K Backgrounds       ART_AND_DESIGN   
23                                Mcqueen Coloring pages       ART_AND_DESIGN   
24                           HD Mickey Minnie Wallpapers       ART_AND_DESIGN   
25                            Harley Quinn wallpapers HD       ART_AND_DESIGN   
26                         Colorfit - Drawing & Coloring       ART_AND_DESIGN   
27                                 Animated Photo Editor       ART_AND_DESIGN   
28                                 Pencil Sketch Drawing       ART_AND_DESIGN   
29                       Easy Realistic Drawing Tutorial       ART_AND_DESIGN   
...                                                  ...                  ...   
10811                                        FR Plus 1.6    AUTO_AND_VEHICLES   
10812                                      Fr Agnel Pune               FAMILY   
10813                                     DICT.fr Mobile             BUSINESS   
10814                               FR: My Secret Pets!                FAMILY   
10815                          Golden Dictionary (FR-AR)  BOOKS_AND_REFERENCE   
10816                                 FieldBi FR Offline             BUSINESS   
10817                               HTC Sense Input - FR                TOOLS   
10818                               Gold Quote - Gold.fr              FINANCE   
10819                                          Fanfic-FR  BOOKS_AND_REFERENCE   
10820                                    Fr. Daoud Lamei               FAMILY   
10821                                            Poop FR               FAMILY   
10822                                          PLMGSS FR         PRODUCTIVITY   
10823                                       List iptv FR        VIDEO_PLAYERS   
10824                                          Cardio-FR              MEDICAL   
10825                                 Naruto & Boruto FR               SOCIAL   
10826          Frim: get new friends on local chat rooms               SOCIAL   
10827                                 Fr Agnel Ambarnath               FAMILY   
10828                            Manga-FR - Anime Vostfr               COMICS   
108

In [137]:
d.shape

(10841, 13)

<h1>Preprocesaminto</h1>

<h2><li> Renombrar Caracteristicas </li></h2>

Para facilitar la implementación del conjunto de datos se arreglaron los nombres y se tradujeron al español. 

In [138]:
d = d.rename(columns={'App':'nombre','Category': 'categoria','Rating':'calificacion','Reviews':'comentarios','Size':'tamanio',
                      'Installs':'instalaciones','Type':'tipo','Price':'precio','Content Rating':'clasificacion',
                      'Genres':'genero','Last Updated':'ult_act','Current Ver':'v_actual','Android Ver':'v_android'})

In [139]:
d.columns

Index([u'nombre', u'categoria', u'calificacion', u'comentarios', u'tamanio',
       u'instalaciones', u'tipo', u'precio', u'clasificacion', u'genero',
       u'ult_act', u'v_actual', u'v_android'],
      dtype='object')

<h2><li> Eliminación de Caracteristicas.</li> </h2>

Siguiendo el objetivo del modelo, son innecesarias las siguientes caracteristicas:
<!--- - instalaciones.-->
- ult_act.
- v_actual.

In [140]:
#del d['instalaciones']
del d['ult_act']
del d['v_actual']
#del d['comentarios']

In [141]:
col = list(d.columns.values)
col

['nombre',
 'categoria',
 'calificacion',
 'comentarios',
 'tamanio',
 'instalaciones',
 'tipo',
 'precio',
 'clasificacion',
 'genero',
 'v_android']

In [142]:
d = d[['nombre',
 'categoria',
 'comentarios',
 'tamanio',
 'instalaciones',
 'tipo',
 'precio',
 'clasificacion',
 'genero',
 'v_android',
 'calificacion']]

In [143]:
d

nombre            categoria  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
5                             Paper flowers instructions       ART_AND_DESIGN   
6                Smoke Effect Photo Maker - Smoke Editor       ART_AND_DESIGN   
7                                       Infinite Painter       ART_AND_DESIGN   
8                                   Garden Coloring Book       ART_AND_DESIGN   
9                          Kids Paint Free - Drawing Fun       ART_AND_DESIGN   
10                               Text on Photo - Fonteee       ART_AND_DESIGN   
11               Name Art Photo Editor - Focus n Filters       ART_AND_DESIGN   
12                        Tattoo Name On My Photo Editor       ART_AND_DESIGN   
13                                 Mandala Coloring Book       ART_AND_DESIGN   
14       3D Color Pixel by Number - Sandbox Art Coloring       ART_AND_DESIGN   
15                       Learn To Draw Kawaii Characters       ART_AND_DESIGN   
16          Photo Designer - Write your name with shapes       ART_AND_DESIGN   
17                              350 Diy Room Decor Ideas       ART_AND_DESIGN   
18                         FlipaClip - Cartoon animation       ART_AND_DESIGN   
19                                          ibis Paint X       ART_AND_DESIGN   
20                           Logo Maker - Small Business       ART_AND_DESIGN   
21             Boys Photo Editor - Six Pack & Men's Suit       ART_AND_DESIGN   
22               Superheroes Wallpapers | 4K Backgrounds       ART_AND_DESIGN   
23                                Mcqueen Coloring pages       ART_AND_DESIGN   
24                           HD Mickey Minnie Wallpapers       ART_AND_DESIGN   
25                            Harley Quinn wallpapers HD       ART_AND_DESIGN   
26                         Colorfit - Drawing & Coloring       ART_AND_DESIGN   
27                                 Animated Photo Editor       ART_AND_DESIGN   
28                                 Pencil Sketch Drawing       ART_AND_DESIGN   
29                       Easy Realistic Drawing Tutorial       ART_AND_DESIGN   
...                                                  ...                  ...   
10811                                        FR Plus 1.6    AUTO_AND_VEHICLES   
10812                                      Fr Agnel Pune               FAMILY   
10813                                     DICT.fr Mobile             BUSINESS   
10814                               FR: My Secret Pets!                FAMILY   
10815                          Golden Dictionary (FR-AR)  BOOKS_AND_REFERENCE   
10816                                 FieldBi FR Offline             BUSINESS   
10817                               HTC Sense Input - FR                TOOLS   
10818                               Gold Quote - Gold.fr              FINANCE   
10819                                          Fanfic-FR  BOOKS_AND_REFERENCE   
10820                                    Fr. Daoud Lamei               FAMILY   
10821                                            Poop FR               FAMILY   
10822                                          PLMGSS FR         PRODUCTIVITY   
10823                                       List iptv FR        VIDEO_PLAYERS   
10824                                          Cardio-FR              MEDICAL   
10825                                 Naruto & Boruto FR               SOCIAL   
10826          Frim: get new friends on local chat rooms               SOCIAL   
10827                                 Fr Agnel Ambarnath               FAMILY   
10828                            Manga-FR - Anime Vostfr               COMICS   


<h2><li> Conversión de Datos</li></h2>

Para poder analizar los datos, los diferentes tipos de datos de cada caracteristica se renombraran con un respectivo numero para poder mapearlos.




In [181]:
a = pd.DataFrame()

In [ ]:
def conversion2(s):
    l = d[s].unique()
    for i in range(len(l)):
        if(l[i] == '')

In [190]:
def conversion(s):
    l = d[s].unique()
    val_antiguo = l
    val_nuevo = range(len(l))
    a = pd.DataFrame()
    a['valor antiguo'] = val_antiguo
    a['valor nuevo'] = val_nuevo
    for i in range(len(l)):
        d.tipo = d.tipo.replace(val_antiguo[i], i)
    return a

<table>
    <th colspan=6><center>categoria</center></th>
    <tr>
        <td>ART_AND_DESIGN</td>
        <td>1</td>
        <td>FAMILY</td>
        <td>12</td>
        <td>PARENTING</td>
        <td>23</td>
    </tr>
    <tr>
        <td>AUTO_AND_VEHICLES</td>
        <td>2</td>
        <td>FINANCE</td>
        <td>13</td>
        <td>PERSONALIZATION</td>
        <td>24</td>
    </tr>
    <tr>
        <td>BEAUTY</td>
        <td>3</td>
        <td>FOOD_AND_DRINK</td>
        <td>14</td>
        <td>PHOTOGRAPHY</td>
        <td>25</td>
    </tr>
    <tr>
        <td>BOOKS_AND_REFERENCE</td>
        <td>4</td>
        <td>GAME</td>
        <td>15</td>
        <td>PRODUCTIVITY</td>
        <td>26</td>
    </tr>
    <tr>
        <td>BUSINESS</td>
        <td>5</td>
        <td>HEALTH_AND_FITNESS</td>
        <td>16</td>
        <td>SHOPPING</td>
        <td>27</td>
    </tr>
    <tr>
        <td>COMICS</td>
        <td>6</td>
        <td>HOUSE_AND_HOME</td>
        <td>17</td>
        <td>SOCIAL</td>
        <td>28</td>
    </tr>
    <tr>
        <td>COMMUNICATION</td>
        <td>7</td>
        <td>LIBRARIES_AND_DEMO</td>
        <td>18</td>
        <td>SPORTS</td>
        <td>29</td>
    </tr>
    <tr>
        <td>DATING</td>
        <td>8</td>
        <td>LIFESTYLE</td>
        <td>19</td>
        <td>TOOLS</td>
        <td>30</td>
    </tr>
    <tr>
        <td>EDUCATION</td>
        <td>9</td>
        <td>MAPS_AND_NAVIGATION</td>
        <td>20</td>
        <td>TRAVEL_AND_LOCAL</td>
        <td>31</td>
    </tr>
    <tr>
        <td>ENTERTAINMENT</td>
        <td>10</td>
        <td>MEDICAL</td>
        <td>21</td>
        <td>VIDEO_PLAYERS</td>
        <td>32</td>
    </tr>
    <tr>
        <td>EVENTS</td>
        <td>11</td>
        <td>NEWS_AND_MAGAZINES</td>
        <td>22</td>
        <td>WEATHER</td>
        <td>33</td>
    </tr>
</table>

<table>
    <th><center>tamanio</center></th>
</table>

<table>
    <th><center>instalaciones</center></th>
</table>

<table>
    <th colspan=4><center>tipo</center></th>
    <tr>
        <td>Free</td>
        <td>0</td>
        <td>Paid</td>
        <td>1</td>
    </tr>
</table>

In [191]:
s = 'tipo'
l = conversion(s)
l

valor antiguo  valor nuevo
0              0            0
1              1            1
2              2            2
3              3            3

In [192]:
d['tipo'].unique()

array([0, 1, 2, 3])

In [126]:
l = d['tipo'].unique()[0]
d.tipo = d.tipo.replace(l, 0)
d.tipo = d.tipo.replace('Paid', 1)

<!-- hay 92 tipos de precios -->
<table>
    <th><center>precio</center></th>
</table>

<table>
    <th colspan=6><center>clasificacion</center></th>
    <tr>
        <td>Adults only 18+</td>
        <td>1</td>
        <td>Everyone 10+</td>
        <td>3</td>
        <td>Teen</td>
        <td>5</td>
    </tr>
    
    <tr>
        <td>Everyone</td>
        <td>2</td>
        <td>Mature 17+</td>
        <td>4</td>
        <td>Unrated</td>
        <td>6</td>
    </tr>
</table>

<table>
    <th colspan=8><center>genero<center></th>
    <tr>
        <td>Action</td> <td>1</td> <td>Casual;Creativity</td> <td>31</td> <td>Finance</td> <td>61</td> <td>Racing;Pretend Play</td> <td>91</td>
    </tr>
    <tr>
        <td>Action;Action & Adventure</td> <td>2</td> <td>Casual;Education</td> <td>32</td> <td>Food & Drink</td> <td>61</td> <td>Role Playing</td> <td>92</td>
    </tr>
    <tr>
        <td>Adventure</td> <td>3</td> <td>Casual;Music & Video</td> <td>33</td> <td>Health & Fitness</td> <td>63</td> <td>Role Playing;Action & Adventure</td> <td>93</td>
    </tr>
    <tr>
        <td>Adventure;Action & Adventure</td> <td>4</td> <td>Casual;Pretend Play</td> <td>34</td> <td>Racing</td> <td>64</td> <td>Role Playing;Brain Games</td> <td>94</td>
    </tr>
    <tr>
        <td>Adventure;Brain Games</td> <td>5</td> <td>Comics</td> <td>35</td> <td>Health & Fitness;Action & Adventure</td> <td>65</td> <td>Role Playing;Education</td> <td>95</td>
    </tr>
    <tr>
        <td>Adventure;Education</td> <td>6</td> <td>Comics;Creativity</td> <td>36</td> <td>Health & Fitness;Education</td> <td>66</td> <td>Role Playing;Pretend Play</td> <td>96</td>
    </tr>
    <tr>
        <td>Arcade</td> <td>7</td> <td>Communication</td> <td>37</td> <td>House & Home</td> <td>67</td> <td>Shopping</td> <td>97</td>
    </tr>
    <tr>
        <td>Arcade;Action & Adventure</td> <td>8</td> <td>Communication;Creativity</td> <td>38</td> <td>Libraries & Demo</td> <td>68</td> <td>Simulation</td> <td>98</td>
    </tr>
    <tr>
        <td>Arcade;Pretend Play</td> <td>9</td> <td>Dating</td> <td>39</td> <td>Lifestyle</td> <td>69</td> <td>Simulation;Action & Adventure</td> <td>99</td>
    </tr>
    <tr>
        <td>Art & Design</td> <td>10</td> <td>Education</td> <td>40</td> <td>Lifestyle;Education</td> <td>70</td> <td>Simulation;Education</td> <td>100</td>
    </tr>
    <tr>
        <td>Art & Design;Action & Adventure</td> <td>11</td> <td>Education;Action & Adventure</td> <td>41</td> <td>Lifestyle;Pretend Play</td> <td>71</td> <td>Simulation;Pretend Play</td> <td>101</td>
    </tr>
    <tr>
        <td>Art & Design;Creativity</td> <td>12</td> <td>Education;Brain Games</td> <td>42</td> <td>Maps & Navigation</td> <td>72</td> <td>Social</td> <td>102</td>
    </tr>
    <tr>
        <td>Art & Design;Pretend Play</td> <td>13</td> <td>Education;Creativity</td> <td>43</td> <td>Medical</td> <td>73</td> <td>Sports</td> <td>103</td>
    </tr>
    <tr>
        <td>Auto & Vehicles</td> <td>14</td> <td>Education;Education</td> <td>44</td> <td>Music</td> <td>74</td> <td>Strategy</td> <td>104</td>
    </tr>
    <tr>
        <td>Beauty</td> <td>15</td> <td>Education;Music & Video</td> <td>45</td> <td>Music & Audio;Music & Video</td> <td>75</td> <td>Strategy;Action & Adventure</td> <td>105</td>
    </tr>
    <tr>
        <td>Board</td> <td>16</td> <td>Education;Pretend Play</td> <td>46</td> <td>Music;Music & Video</td> <td>76</td> <td>Strategy;Creativity</td> <td>106</td>
    </tr>
    <tr>
        <td>Board;Action & Adventure</td> <td>17</td> <td>Educational</td> <td>47</td> <td>News & Magazines</td> <td>77</td> <td>Strategy;Education</td> <td>107</td>
    </tr>
    <tr>
        <td>Board;Brain Games</td> <td>18</td> <td>Educational;Action & Adventure</td> <td>48</td> <td>Parenting</td> <td>78</td> <td>Tools</td> <td>108</td>
    </tr>
    <tr>
        <td>Board;Pretend Play</td> <td>19</td> <td>Educational;Brain Games</td> <td>49</td> <td>Parenting;Brain Games</td> <td>79</td> <td>Tools;Education</td> <td>109</td>
    </tr>
    <tr>
        <td>Books & Reference</td> <td>20</td> <td>Educational;Creativity</td> <td>50</td> <td>Parenting;Education</td> <td>80</td> <td>Travel & Local</td> <td>110</td>
    </tr>
    <tr>
        <td>Books & Reference;Creativity</td> <td>21</td> <td>Educational;Education</td> <td>51</td> <td>Parenting;Music & Video</td> <td>81</td> <td>Travel & Local;Action & Adventure</td> <td>111</td>
    </tr>
    <tr>
        <td>Books & Reference;Education</td> <td>22</td> <td>Educational;Pretend Play</td> <td>52</td> <td>Personalization</td> <td>82</td> <td>Trivia</td> <td>112</td>
    </tr>
    <tr>
        <td>Business</td> <td>23</td> <td>Entertainment</td> <td>53</td> <td>Photography</td> <td>83</td> <td>Trivia;Education</td> <td>113</td>
    </tr>
    <tr>
        <td>Card</td> <td>24</td> <td>Entertainment;Action & Adventure</td> <td>54</td> <td>Productivity</td> <td>84</td> <td>Video Players & Editors</td> <td>114</td>
    </tr>
    <tr>
        <td>Card;Action & Adventure</td> <td>25</td> <td>Entertainment;Brain Games</td> <td>55</td> <td>Puzzle</td> <td>85</td> <td>Video Players & Editors;Creativity</td> <td>115</td>
    </tr>
    <tr>
        <td>Card;Brain Games</td> <td>26</td> <td>Entertainment;Creativity</td> <td>56</td> <td>Puzzle;Action & Adventure</td> <td>86</td> <td>Video Players & Editors;Music & Video</td> <td>116</td>
    </tr>
    <tr>
        <td>Casino</td> <td>27</td> <td>Entertainment;Education</td> <td>57</td> <td>Puzzle;Brain Games</td> <td>87</td> <td>Weather</td> <td>117</td>
    </tr>
    <tr>
        <td>Casual</td> <td>28</td> <td>Entertainment;Music & Video</td> <td>58</td> <td>Puzzle;Creativity</td> <td>88</td> <td>Word</td> <td>118</td>
    </tr>
    <tr>
        <td>Casual;Action & Adventure</td> <td>29</td> <td>Entertainment;Pretend Play</td> <td>59</td> <td>Puzzle;Education</td> <td>89</td> <td></td> <td>119</td>
    </tr>
    <tr>
        <td>Casual;Brain Games</td> <td>30</td> <td>Events</td> <td>60</td> <td>Racing;Action & Adventure</td> <td>90</td> <td></td> <td>120</td>
    </tr>
</table>

In [158]:
conversion('genero')

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo', 'Lifestyle',
       'Lifestyle;Pretend Play', 'Adventure;Action & Adventure', 'Arcade',
       'Casual', 'Card', 'Casual;Pretend Play', 'Action', 'Strategy',
       'Puzzle', 'Sports', 'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role Playing

<table>
    <th><center>v_android</center></th>
</table>

In [144]:
d.groupby('tipo').agg(['mean','std'])

calificacion          
             mean       std
tipo                       
0       19.000000       NaN
Free     4.186203  0.512338
Paid     4.266615  0.547523

In [163]:
l = d['tipo'].unique()
print l
len(l)

['Free' 'Paid' nan '0']


4

In [208]:
d[d.calificacion.isnull()]

nombre            categoria  \
23                                Mcqueen Coloring pages       ART_AND_DESIGN   
113                            Wrinkles and rejuvenation               BEAUTY   
123                               Manicure - nail design               BEAUTY   
126                         Skin Care and Natural Beauty               BEAUTY   
129                  Secrets of beauty, youth and health               BEAUTY   
130                   Recipes and tips for losing weight               BEAUTY   
134                        Lady adviser (beauty, health)               BEAUTY   
163                           Anonymous caller detection  BOOKS_AND_REFERENCE   
180                  SH-02J Owner's Manual (Android 8.0)  BOOKS_AND_REFERENCE   
185                       URBANO V 02 instruction manual  BOOKS_AND_REFERENCE   
227                                       Y! Mobile menu             BUSINESS   
321    【Ranobbe complete free】 Novelba - Free app tha...               COMICS   
478                                    Truth or Dare Pro               DATING   
479      Private Dating, Hide App- Blue for PrivacyHider               DATING   
480                                 Ad Blocker for SayHi               DATING   
610                                    Random Video Chat               DATING   
613                 Random Video Chat App With Strangers               DATING   
617             Meet With Strangers: Video Chat & Dating               DATING   
620             Ost. Zombies Cast - New Music and Lyrics               DATING   
621                                   Dating White Girls               DATING   
623                                         Geeks Dating               DATING   
624                          Live chat - free video chat               DATING   
626                  Fishing Brain & Boating Maps Marine               DATING   
627                                   CAM5678 Video Chat               DATING   
628                              Video chat live advices               DATING   
629                                       chat live chat               DATING   
630                                    Pet Lovers Dating               DATING   
631            Friend Find: free chat + flirt dating app               DATING   
632                                         Latin Dating               DATING   
635                                          Wifi Mingle               DATING   
...                                                  ...                  ...   
10746                                         FP Opgaver                TOOLS   
10748                                            FP Live        COMMUNICATION   
10751                                          FP Market               FAMILY   
10759                                     FP NFC Rewrite                TOOLS   
10761                             Greek Bible FP (Audio)  BOOKS_AND_REFERENCE   
10762                                      The FP Shield   NEWS_AND_MAGAZINES   
10764                                  FP Transportation    AUTO_AND_VEHICLES   
10769                                        FQ Magazine            LIFESTYLE   
10772                     FQ Load Board for Transporters             BUSINESS   
10773                                           FQ India            LIFESTYLE   
10774                                            Miss FQ   NEWS_AND_MAGAZINES   
10775                                 FQ - Football Quiz               SPORTS   
10788    Fountain Live Wallpaper HD – Dubai Wallpaper 3D      PERSONALIZATION   
10794                                            PopStar               FAMILY   
10798                               Word Search Tab 1 FR               FAMILY   
10806                                    SnakeBite911 FR              MEDICAL   
10807                                          My FR App                TOOLS   
10808                                 lesparticuliers.fr            LIFESTYLE   


In [146]:
d[d.genero == 'Books & Reference;Creativity']

nombre categoria comentarios tamanio instalaciones  tipo precio  \
7629  Wuwu & Co.    FAMILY           9     77M          100+  Paid  $2.99   

     clasificacion                        genero   v_android  calificacion  
7629      Everyone  Books & Reference;Creativity  4.1 and up           NaN

In [147]:
d[d.categoria == '1.9']

nombre categoria comentarios tamanio  \
10472  Life Made WI-Fi Touchscreen Photo Frame       1.9        3.0M  1,000+   

      instalaciones tipo    precio clasificacion             genero v_android  \
10472          Free    0  Everyone           NaN  February 11, 2018       NaN   

       calificacion  
10472          19.0

In [80]:
a= pd.DataFrame(np.array([['ola',1,0],[3,3,3],[1,3,3],[0,1,3],[3,1,2]]), columns=["col1", "col2", "col3"])

In [81]:
a

col1 col2 col3
0  ola    1    0
1    3    3    3
2    1    3    3
3    0    1    3
4    3    1    2

In [82]:
a.col1 = a.col1.replace(['ola',8],[0, 'ola'])

In [83]:
a

col1 col2 col3
0    0    1    0
1    3    3    3
2    1    3    3
3    0    1    3
4    3    1    2

In [55]:
a[a.col1 == 8]

col1 col2 col3
0    8    1    0

In [16]:
x = a.values[:,:-1]

x

array([[0, '1'],
       ['3', '3'],
       ['1', '3'],
       ['0', '1'],
       ['3', '1']], dtype=object)

In [18]:
y = a.values[:,-1]
y

array(['0', '3', '3', '3', '2'], dtype=object)

In [20]:
c = a.values[:3]
c

array([[0, '1', '0'],
       ['3', '3', '3'],
       ['1', '3', '3']], dtype=object)

In [23]:
y[-3:]

array(['3', '3', '2'], dtype=object)

In [ ]:
pd.Series(pd.Categorical(list('baabc'))).unique()

In [133]:
pd.Series(pd.Categorical(list('baabc'), ordered=True)).unique()

[b, a, c]
Categories (3, object): [a < b < c]